src/components/GoalPreview/GoalPreview.tsx

```diff
@@ -217,9 +218,15 @@ const GoalPreview: React.FC<GoalPreviewProps> = ({ preview, onClose, onDelete })
 
                     <StyledImportantActions>
                         <StyledPublicActions>
-                            {nullable(goal?.state, (s) => (
-                                <StateSwitch state={s} flowId={goal?.project?.flowId} onClick={onGoalStateChange} />
-                            ))}
+                            {goal?._isEditable
+                                ? nullable(goal?.state, (s) => (
+                                      <StateSwitch
+                                          state={s}
+                                          flowId={goal?.project?.flowId}
+                                          onClick={onGoalStateChange}
+                                      />
+                                  ))
+                                : nullable(goal?.state, (s) => <State title={s.title} hue={s.hue} />)}
```

Comment:
you probably wrap this into `nullable` alltogether or use two statements like `nullable(goal?.state && goal.?_isEditable)` so that you don't have multiple logic in a trenary